### PR TITLE
Implements Mech Coins

### PR DESCRIPTION
**Pull request by Ben10083**
## About The Pull Request
Mech computers now require a mech coin to be inserted to assemble a mech, MPs spawn with one, but must get more from req (250 points, point cost subject to change based on feedback)
## Why It's Good For The Game
We are already seeing that some pilots are utilizing the fact mechs can be reassembled at not cost in unintended ways, such as bringing multiple mechs groundside at once and switching in and out of them if one gets damaged. We also seen that mech pilots, if they survive when mech is destroyed, can easily just head shipside and just assemble a new mech, undoing xeno progress.

With this pr, that will no longer be the case. The first mech is free, other ones require req points, to make losing a mech a significant setback. They are given the ability to test out mechs in Valhalla, they should be expected to stick with the design they chose originally, like how CAS is.
## Changelog
:cl:
add: Mech Coin added to vehicle tab in requisitions.
add: Mech pilot spawns with Mech Coin
balance: Mech Computer now requires the user to insert a mech coin before assembly.
imageadd: Mech Coin sprite added
/:cl:

### CODE REVIEW

**Comment by Ben10083** [comment]
image

**Comment by Coastfront** [comment]
Special shoutout to when mechs were designed, failing to see the disparity between the fact that for some reason, xenos, lose a shitton of momentum and must expend time to regain a queen, whereas marines, who potentially get multiple mechs, could simply regenerate damaged mechs for the grand cost of about a minute or so
where xenos spent a walletfull marines spent a dime.

This makes it so that marines tangibly lose rescources as penalty for an mp fucking up
Good.

**Comment by Starfirejordan** [comment]
Posted this in the discord about my concerns, gonna post them here aswell 

ill list 3 reason why I think 300 is over priced
and it should ATLEAST be lowered to something like 200 - 250

1) mech are always at the front of the push thus take the most damage and punishment, Light mech would not only be FORCED to use a IFF weapon to not risk getting trapped and kill thus costing 300 to be replaced when it was simply doing its job and getting punished heavily for doing it

2) 300 points? any RO would buy req gear for skilled marine statics before EVER getting mech pilot a mech coin i could get 3 Scout rifles with that amount of req point with some ammo and deal more damage then most mech guns AND have IFF not to mention how I have to give SG ammo or buying metal for engi ETC... it the moment it more worth buy shit for marines then getting a single person a new mech

3) it would over all ruin the fun for the person using the mech, as if they slip up once, then there out of the fight, ive gone many round where we had jack shit for req points, seen plenty of times where no engi fixes miners etc... if there isnt any marines doing what they need todo and you mess up and get revived, all you are is just a marine

**Comment by Ben10083** [comment]
> Posted this in the discord about my concerns, gonna post them here aswell beside that, good PR makes fighting mechs as xeno less bullshit
> 
> ill list 3 reason why I think 300 is over priced and it should ATLEAST be lowered to something like 200 - 250
> 
> 1. mech are always at the front of the push thus take the most damage and punishment, Light mech would not only be FORCED to use a IFF weapon to not risk getting trapped and kill thus costing 300 to be replaced when it was simply doing its job and getting punished heavily for doing it
> 2. 300 points? any RO would buy req gear for skilled marine statics before EVER getting mech pilot a mech coin i could get 3 Scout rifles with that amount of req point with some ammo and deal more damage then most mech guns AND have IFF not to mention how I have to give SG ammo or buying metal for engi ETC... it the moment it more worth buy shit for marines then getting a single person a new mech
> 3. it would over all ruin the fun for the person using the mech, as if they slip up once, then there out of the fight, ive gone many round where we had jack shit for req points, seen plenty of times where no engi fixes miners etc... if there isnt any marines doing what they need todo and you mess up and get revived, all you are is just a marine

1 is just a risk with mechs, you are priority target for xenos, not much can be done about that.
2 I accept as valid, but will wait for more feedback before determining any change in price
3 is a scenario TAD players can face just as well, if the PO abandons the TAD due to it being FUBAR they are basically now marines (alamo doesnt count fuck off), MPs can at least BUY a new mech.

**Comment by Al-1ce** [comment]
300 is overkill, if not insane. By the time req has that much cash to burn, the round is already over. 150-100 would be reasonable.

**Comment by Ben10083** [comment]
> 300 is overkill, if not insane. By the time req has that much cash to burn, the round is already over. 150-100 would be reasonable.

150-100 is about b18 level, which is too low. 300 is likely too high though, likely will bring it down to 250 when I am back at my computer

File: code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm [hunk 1]
```diff
@@ -84,11 +84,20 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
 	QDEL_LIST(plane_masters)
 	return ..()
 
+/obj/item/coin/mech
+	name = "Mech Coin"
```
Comment (XSlayer300) [inline code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm:88]:
```suggestion
	name = "mech coin"
```

File: code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm [hunk 2]
```diff
@@ -84,11 +84,20 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
...
+	desc = "This coin is needed to assemble a mech, don't lose this."
```
Comment (XSlayer300) [inline code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm:89]:
```suggestion
	desc = "An encrypted data coin needed to assemble a mech. Don't lose this."
```

**Comment by FosterGrime-Heymynameslouie** [comment]
this is very good r on how to stop mass amounts of mechs so mps can hope in and out, but

is it possible to make mechs refundable? Like throw em in the req pit to get back 300 or 250, whichever gets chosen amount of points, personally because I've seen other people and I myself have accidentally made wrong loadouts, misclicked the wrong module I wanted, got the wrong gun etc. One of the main reasons why there's one or two mechs near the console itself because of wrong loadout. other than that this is cool! 
Pls im stupid

spedde mgehr

**Comment by iLikeMemes** [comment]
> is it possible to make mechs refundable? Like throw em in the req pit to get back 300 or 250, whichever gets chosen amount of points, personally because I've seen other people and I myself have accidentally made wrong loadouts, misclicked the wrong module I wanted, got the wrong gun etc. One of the main reasons why there's one or two mechs near the console itself because of wrong loadout. 

You already mentioned people just tossing the mech/coin in the req pit for points but this could be circumvented by not giving points for the coin or mech but just spawning a crate with a mech coin once one a mech is sent down the req elevator. No points given in that way.

**Comment by FosterGrime-Heymynameslouie** [comment]
> > is it possible to make mechs refundable? Like throw em in the req pit to get back 300 or 250, whichever gets chosen amount of points, personally because I've seen other people and I myself have accidentally made wrong loadouts, misclicked the wrong module I wanted, got the wrong gun etc. One of the main reasons why there's one or two mechs near the console itself because of wrong loadout.
> 
> You already mentioned people just tossing the mech/coin in the req pit for points but this could be circumvented by not giving points for the coin or mech but just spawning a crate with a mech coin once one a mech is sent down the req elevator. No points given in that way.

"Spawning a mech point whenever a mech is throw down into the pit instead of 300 points for a refund"
Yes you are a genuis, I didnt even think of this
this please^^^

**Comment by zzurge** [comment]
300 points? IN THIS ECONOMY?

**Comment by Ben10083** [comment]
After feedback, cost reduced to 250, This is just over the cost of two b18s.

**Comment by TiviPlus** [comment]
Not interested

Instead you may do the following in the same order:
Add UI customuization updating
Add timer (ie only 1 mech every 15 minutes or something like that per person using it)

**Pull request closed without merging**